go-controller/pkg/ovn/policy.go

```diff
@@ -247,6 +247,21 @@ func (oc *Controller) syncNetworkPolicies(networkPolicies []interface{}) error {
 	if err != nil {
 		return fmt.Errorf("cannot find stale arp allow ACLs: %v", err)
 	}
+	// Remove these stale ACLs from port groups and then delete them
+	for _, gressACL := range gressACLs {
+		gressACL := gressACL
+		pgName := strings.Replace(gressACL.Match, "inport == @", "", 1)
+		pgName = strings.Replace(pgName, " && arp", "", 1)
```

Comment:
this doesn't really look safe, even though it's the most performant solution
Maybe we can just filter all in/e gress port groups whose name ends with `ingressDefaultDenySuffix` or `egressDefaultDenySuffix` and for all of them run `DeleteACLsFromPortGroupOps` that should work even if ACL is not referenced by this port group?
From the performance POV - this should only run once when stale ACLs are still present and filtering port groups should also help a little bit here. wdyt?

Reply:
yes I think we are ok here because its a one time thing during upgrade. I didn't want to do another pull of all default port groups. I can write comments there to make the logic clear if needed.

Reply:
yeah let's maybe just add the example of how this match looks like in the comment for ingress and egress acls?